examples/servers/simple-streamablehttp-stateless/mcp_simple_streamablehttp_stateless/server.py

```diff
@@ -131,6 +132,14 @@ async def lifespan(app: Starlette) -> AsyncIterator[None]:
         ],
         lifespan=lifespan,
     )
+    
+    # Add CORS middleware to expose Mcp-Session-Id header for browser-based clients
+    starlette_app.add_middleware(
+        CORSMiddleware,
+        allow_origins=["*"],  # Allow all origins - adjust as needed for production
+        allow_methods=["GET", "POST", "DELETE"],  # MCP streamable HTTP methods
+        expose_headers=["Mcp-Session-Id"],
+    )
```

Comment:
You need to wrap the whole application instead of using `add_middleware`, otherwise 500s don't get in this.